### PR TITLE
Add non-functional requirements into the developer guide

### PR DESCRIPTION
**Pull request by Nicholas-Cheng-De-Fei**
Close the 2 issues:
- https://github.com/AY2425S1-CS2103-F10-3/tp/issues/78
- https://github.com/AY2425S1-CS2103-F10-3/tp/issues/93

### CODE REVIEW

**Review by ITLimJiaWei — CHANGES_REQUESTED** [review]
There is a jump from NFR 7 to 9, rename NFR 9 to 8. Also for NFR 6, do you mean full functionality? or those save file error cases are handled appropriately?

**Comment by Nicholas-Cheng-De-Fei** [comment]
For NFR 6 I am just talking about in the event of a save file corruption or mission only. If this happens it should be handled accordingly

**Review by ITLimJiaWei — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -317,9 +317,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `17` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1.  The should work on any _mainstream OS_ as long as it has Java `17` or above installed.
+2.  Should be able to hold up to 1000 contacts without a noticeable sluggishness in performance for typical usage.
+3.  The should be able to maintain data integrity, where system should ensure that data is accurate and consistent, preventing duplicate entries.
```
Comment (KennethTeo2002) [inline docs/DeveloperGuide.md:322]: Is there missing words "The system should..." for these 3 lines?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -317,9 +317,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+4.  The system should be developed in a modular way for easier updates and bug fixes.
+5.  The system should be able to ensure data consistency accross all instances.
+6.  The system should be able to continue functioning in the event of a missing or corrupted save file.
+7.  The system should be able to encrypt sensitive data.
```
Comment (KennethTeo2002) [inline docs/DeveloperGuide.md:326]: I think "be able to" should be removed?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Nicholas-Cheng-De-Fei) [reply docs/DeveloperGuide.md]: I have made the neecessary changes

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Nicholas-Cheng-De-Fei) [reply docs/DeveloperGuide.md]: I have removed the "be able to"

**Review by KennethTeo2002 — APPROVED** [review]
LGTM

**Pull request merged**